quickstart/dotnet/Redistest/Program.cs

```diff
@@ -1,87 +1,89 @@
     }
 
-    class Program
+    internal class Program
     {
-        private static RedisConnection _redisConnection;
-
         static async Task Main(string[] args)
         {
-            _redisConnection = await RedisConnection.InitializeAsync(redisHostName: ConfigurationManager.AppSettings["RedisHostName"].ToString());
+            // Connection string should be the cache host name and port e.g. "cachename.redis.azure.net:10000"
+            var connectionString = ConfigurationManager.AppSettings["RedisConnectionString"].ToString();
+
+            // Connect to the cache using Azure credentials loaded from the current environment. For details see https://learn.microsoft.com/dotnet/azure/sdk/authentication
+            // One approach is to assign yourself to an access policy on the cache (see https://aka.ms/redis/entra-auth), and ensure
+            // that you're logged in with the same account in Visual Studio (see File -> Account Settings...)
+            // For other authentication methods see https://github.com/Azure/Microsoft.Azure.StackExchangeRedis
+            var configurationOptions = await ConfigurationOptions.Parse(connectionString).ConfigureForAzureWithTokenCredentialAsync(new DefaultAzureCredential());
+            configurationOptions.AbortOnConnectFail = true; // Remove this option for code in production. It makes connections less resilient, but easier to diagnose connection problems while debugging.
 
-            try
+            Console.WriteLine($"Connecting to '{connectionString}'...");
+            using (var redisConnection = await ConnectionMultiplexer.ConnectAsync(configurationOptions))
             {
-                // Perform cache operations using the cache object...
-                Console.WriteLine("Running... Press any key to quit.");
+                Console.WriteLine($"Connected successfully!");
+                Console.WriteLine();
 
-                while (!Console.KeyAvailable)
-                {
-                    Task thread1 = Task.Run(() => RunRedisCommandsAsync("Thread 1"));
-                    Task thread2 = Task.Run(() => RunRedisCommandsAsync("Thread 2"));
+                var redis = redisConnection.GetDatabase();
 
-                    Task.WaitAll(thread1, thread2);
-                }
-            }
-            finally
-            {
-                _redisConnection.Dispose();
+                Console.WriteLine("Executing Redis commands on two threads to demonstrate how a single Redis connection can safely multiplex commands from many concurrent threads...");
+                Console.WriteLine();
+
+                var thread1 = RunRedisCommandsAsync("Thread1", redis);
+                var thread2 = RunRedisCommandsAsync("Thread2", redis);
+
+                await Task.WhenAll(thread1, thread2);
             }
         }
 
-        private static async Task RunRedisCommandsAsync(string prefix)
+        private static async Task RunRedisCommandsAsync(string threadName, IDatabase redis)
```

Comment:
Should the first line of this method be `await Task.Yield()`? To guarantee/force evaluation on different thread pool threads?

Reply:
I don't think so... we wouldn't want customers switching threads any more than necessary in their flow. They may want to fork off at a higher level, but I think that's outside the scope of quickstarts.